src/providers/ogr/qgsgeopackagedataitems.cpp

```diff
@@ -400,24 +415,34 @@ bool QgsGeoPackageConnectionItem::handleDrop( const QMimeData *data, Qt::DropAct
 
             QgsApplication::taskManager()->addTask( exportTask.release() );
           }
-        }
-        else
-        {
-          importResults.append( tr( "%1: Not a valid layer!" ).arg( u.name ) );
-          hasError = true;
-        }
-      }
-    }
-    else
-    {
-      // TODO: implement raster import
-      QgsMessageOutput *output = QgsMessageOutput::createMessageOutput();
-      output->setTitle( tr( "Import to GeoPackage database failed" ) );
-      output->setMessage( tr( "Failed to import some layers!\n\n" ) + QStringLiteral( "Raster import is not yet implemented!\n" ), QgsMessageOutput::MessageText );
-      output->showMessage();
-    }
+          else  // Import raster
+          {
+            // In case we need it
+            // QgsRasterLayer* rasterSrcLayer = dynamic_cast < QgsRasterLayer* > ( srcLayer );
+
+            const char *args[] = { "-of", "gpkg", "-co", QStringLiteral( "RASTER_TABLE=%1" ).arg( u.name ).toUtf8().constData(), "-co", "APPEND_SUBDATASET=YES", nullptr };
+            GDALTranslateOptions *psOptions = GDALTranslateOptionsNew( ( char ** )args, nullptr );
+            GDALDatasetH hSrcDS = GDALOpen( u.uri.toUtf8().constData(), GA_ReadOnly );
+            CPLErrorReset();
+            GDALDatasetH hOutDS = GDALTranslate( mPath.toUtf8().constData(), hSrcDS, psOptions, NULL );
```

Comment:
GDALTranslate() requires GDAL >= 2.1. Currently only >= 2.0 is required : https://github.com/qgis/QGIS/blob/master/cmake/FindGDAL.cmake#L104